lib/widgets/inbox.dart

```diff
@@ -0,0 +1,518 @@
+      }
+      dmItems.add((dmNarrow, countInNarrow));
+      allDmsCount += countInNarrow;
+    }
+    if (allDmsCount > 0) {
+      sections.add(AllDmsSectionData(allDmsCount, dmItems));
+    }
+
+    for (final MapEntry(key: streamId, value: topics) in unreadsModel!.streams.entries) {
+      if (!store.subscriptions.containsKey(streamId)) {
+        // Filter out any straggling unreads in unsubscribed streams.
+        // There won't normally be any, but it happens with certain infrequent
+        // state changes, typically for less than a few hundred milliseconds.
+        // See [Unreads].
+        //
+        // Also, we want to depend on the subscription data for things like
+        // the stream color.
+        continue;
+      }
+      final topicItems = <(String, int)>[];
+      int countInStream = 0;
+      for (final MapEntry(key: topic, value: messageIds) in topics.entries) {
+        final countInTopic = messageIds.length;
+        topicItems.add((topic, countInTopic));
+        countInStream += countInTopic;
+      }
+      if (countInStream == 0) {
+        continue;
+      }
+      sections.add(StreamSectionData(streamId, countInStream, topicItems));
+    }
+
+    // TODO(#346) Filter out muted messages
+
+    return Scaffold(
+      appBar: AppBar(title: const Text('Inbox')),
+      body: StickyHeaderListView.builder(
+        itemCount: sections.length,
+        itemBuilder: (context, index) {
+          final section = sections[index];
+          switch (section) {
+            case AllDmsSectionData(:var count):
+              final header = AllDmsHeaderItem(
+                count: count,
+                collapsed: allDmsCollapsed,
+                pageState: this,
+              );
+              return StickyHeaderItem(header: header,
+                child: AllDmsSection(
+                  data: section,
+                  collapsed: allDmsCollapsed,
+                  pageState: this,
+                ));
+            case StreamSectionData(:var streamId, :var count):
+              final collapsed = collapsedStreamIds.contains(streamId);
+              final header = StreamHeaderItem(
+                streamId: streamId,
+                count: count,
+                collapsed: collapsed,
+                pageState: this,
+              );
+              return StickyHeaderItem(header: header,
+                child: StreamSection(data: section, collapsed: collapsed, pageState: this));
+          }
+        }));
+  }
+}
+
+sealed class InboxSectionData {
+  const InboxSectionData();
+}
+
+class AllDmsSectionData extends InboxSectionData {
+  final int count;
+  final List<(DmNarrow, int)> items;
+
+  const AllDmsSectionData(this.count, this.items);
+}
+
+class StreamSectionData extends InboxSectionData {
+  final int streamId;
+  final int count;
+  final List<(String, int)> items;
+
+  const StreamSectionData(this.streamId, this.count, this.items);
+}
+
+class AllDmsHeaderItem extends StatelessWidget {
```

Comment:
Can these widget classes be private?

If they remain public, they should get more specific names that make it clear they're for the inbox. E.g., `InboxAllDmsHeaderItem`. That will mitigate having them show up in IDE autocomplete when working in unrelated parts of the code.